core/src/services/s3/backend.rs

```diff
@@ -1027,14 +1037,24 @@ impl Access for S3Backend {
     }
 
     async fn list(&self, path: &str, args: OpList) -> Result<(RpList, Self::Lister)> {
-        let l = S3Lister::new(
-            self.core.clone(),
-            path,
-            args.recursive(),
-            args.limit(),
-            args.start_after(),
-        );
-        Ok((RpList::default(), oio::PageLister::new(l)))
+        let l = if self.core.bucket_versioning_enabled && args.version() {
```

Comment:
We need to raise an error if `version` is enabled but the bucket doesn't support it.